plat/drivers/gic/gic-v2.c

```diff
@@ -179,11 +179,17 @@ void gic_sgi_gen_to_list(uint32_t sgintid, uint8_t targetlist)
 	unsigned long irqf;
 
 	/* spin lock here is needed when smp is supported */
-	dist_lock(gicv2_drv);
```

Comment:
@rene Done. The problematic commit is the third one, SGIs don't reach their destination when using gic-v3. You can see this when running the test for my SMP pull request [1], the application hangs forever because the `unlock()` in the interrupt handler is never reached [2].

[1]: https://github.com/unikraft/unikraft/pull/373
[2]:https://github.com/unikraft/unikraft/pull/373/files#diff-47b8b12220e68c052272b4198ee33f962e5c9f8cf63b95fda9b47c0ac4fde9d7R70

Reply:
@razvanvirtan looks good to me. According to your last comment, so basically SGI generation is not working on gic-v3? I will give a try in your SMP branch...

Reply:
Yes, when using gic-v3 with my SMP bootcode the application blocks.

Reply:
Also, note that you will have to select uklock when testing the SMP draft, I've added this in the SMP PR too (I think I will update the PR so that uklock is included by default when SMP is enabled).